### PR TITLE
feat(ARCH-313): ui-scripts 7.0

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
**What is the problem this PR is trying to solve?**

Our current build create few but big bundles
This is not sustainable.

**What is the chosen solution to this problem?**

Lets build our projects by using CDN. For that the build now rely on UMD format for libraries.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: babel/plugin-import-from-index/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "@talend/babel-plugin-import-from-index",
+  "version": "1.0.0",
+  "description": "",
+  "main": "lib/index.js",
+  "scripts": {
+    "test": "jest ./src/index.test.js",
+    "build": "babel src --out-dir build --ignore '**/*.test.js'"
```
Comment (jsomsanith-tlnd) [inline babel/plugin-import-from-index/package.json:8]: Do we really need the build step and the `babel.config.js` file ?

File: packages/config-react-webpack/config/webpack.config.js [hunk 2]
```diff
@@ -190,6 +190,12 @@ module.exports = ({ getUserConfig, mode }) => {
 		resolve: {
 			extensions: ['.js', useTypescript && '.ts', useTypescript && '.tsx'].filter(Boolean),
 		},
+		externals: {
+			react: 'React',
+			'react-dom': 'ReactDOM',
+			'@talend/react-components': 'TalendReactComponents',
+			'react-i18next': 'ReactI18next',
+		},
```
Comment (jsomsanith-tlnd) [inline packages/config-react-webpack/config/webpack.config.js:198]: To enable onPremise builds, we should make that customisable, perhaps just a flag from talend-scripts.json.
- the externals flag should be true by default
- (remark) depending on the flag we need to add this `externals` list, but we need to think about how to we sync that list with the script that will insert the scripts tags to load via CDN

File: babel/plugin-import-from-index/package.json (hunk 1 above)
Comment (frassinier) [reply babel/plugin-import-from-index/package.json:8]: Safer I guess

File: packages/config-react-webpack/config/webpack.config.js (hunk 2 above)
Comment (frassinier) [reply packages/config-react-webpack/config/webpack.config.js:198]: Sure, let's do a POC with this at first?
👀 and then add the flag for on prem delivery process

File: babel/plugin-import-from-lib/package.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@talend/babel-plugin-plugin-import-from-lib",
+  "name": "@talend/babel-plugin-import-from-lib",
```
Comment (jmfrancois) [inline babel/plugin-import-from-lib/package.json]: 🤦

File: package.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
   "devDependencies": {
+    "cpx2": "^2.0.0",
```
Comment (jmfrancois) [inline package.json:3]: drop this

File: packages/config-babel/package.json [hunk 5]
```diff
@@ -21,9 +21,11 @@
     "@babel/plugin-transform-object-assign": "^7.8.3",
     "@babel/preset-env": "^7.8.7",
     "@babel/preset-react": "^7.8.3",
+    "@talend/babel-plugin-import-from-index": "^1.4.3",
+    "@talend/babel-plugin-import-from-lib": "^0.0.0",
```
Comment (jmfrancois) [inline packages/config-babel/package.json:25]: drop this one

File: packages/config-ng-webpack/config/webpack.config.js [hunk 6]
```diff
@@ -18,7 +19,7 @@ module.exports = {
 							explicitOnly: false,
 						},
 					},
-					{ loader: 'babel-loader' },
+					{ loader: 'babel-loader', options: getBabelConfig() },
```
Comment (jmfrancois) [inline packages/config-ng-webpack/config/webpack.config.js:22]: backport this on 6.x

**Review by jmfrancois — APPROVED** [review]

**Pull request merged**